frontend/catalyst/pennylane_extensions.py

```diff
@@ -1289,6 +1289,69 @@ def conditional_fn():
             return "if provided, return types need to be identical in both branches"
 
         ret_val = conditional_fn()  # must invoke the defined function
+
+    .. details::
+        :title: Usage details
+        :href: usage-details
```

Comment:
Yep, I think we should start adding more details where needed to docstrings, because as a user I always find myself going to docstrings for reference while working!